compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -1705,6 +1690,115 @@ impl<'block> BrilligBlock<'block> {
         }
     }
 
+    fn initialize_constant_array(
+        &mut self,
+        data: &im::Vector<ValueId>,
+        typ: &Type,
+        dfg: &DataFlowGraph,
+        pointer: MemoryAddress,
+    ) {
+        if data.is_empty() {
+            return;
+        }
+        let item_types: &Vec<_> = match typ {
+            Type::Array(item_types, _) => item_types.borrow(),
+            Type::Slice(item_types) => item_types.borrow(),
+            _ => unreachable!("ICE: Array constant is not of array or slice type {typ:?}"),
+        };
```

Comment:
Can you use `.element_types()` here or move this code into a method on Type?

Reply:
`element_types()` was fine, but had to clone since it consumes the `Type`

Reply:
I think creating a new method on Type to avoid cloning would have been better but I won't block over it